app/controllers/admin/users_controller.rb

```diff
@@ -83,6 +83,26 @@ def activate
     end
   end
 
+  # allow an admin to sign-in as any other user
+
+  def switch_to_user
+    if current_user != @user
+      sign_in(:user, @user, { bypass: true })
+      session[:original_admin_user_id] = current_user.id
```

Comment:
I guess `sign_in` doesn't set `current_user`, but it might feel safer as:

```ruby
old_user = current_user
sign_in(:user, @user, { bypass: true })
session[:original_admin_user_id] = old_user.id
```